### DISCUSSION TITLE
Schmid Factor > 0.5

### DISCUSSION THREAD

**Discussion by Ronit-Roy** [mtex-toolbox/mtex#1801] [Ask Anything] [answered]
Hi,

I need some help with calculating the Schmid factor for HCP systems. I'm getting SF values of more than 0.5 for specific stress tensors obtained by FEA. I calculated in both crystal and specimen coordinates and ended up with the same results. However, if I reduce/normalize the stress tensor, the resultant Schmid factor also reduces proportionally. As discussed in this forum, I used the Frobenius norm to normalize the stress tensor. But that didn't work for me. Is there any other specific normalization method standard for the Schmid factor analysis with complex stress tensors? I'm attaching my script and ref files here. Could you help me with this?  

Many thanks,
Ronit Roy

`% crystal symmetry
CS = crystalSymmetry('6/mmm', [3.232 3.232 5.148], 'X||a*', 'Y||b', 'Z||c*',...
	'mineral', 'Zirconium - alpha', 'color', 'light blue');

% Load grain mean orienatation (Zirconium)
load('MeanOrientation.mat');

% Load Stress tensor
load('Stress.mat');

%%%%%%%%%%%%%%%%% Analysis in Specimen coordinate%%%%%%%%%%%%%%%%%%%%%%%
sS=slipSystem.prismaticA(CS,1);					% CRSS=1
sS=sS.symmetrise('antipodal');

% Rotate slip systems to global coordinates
sSLocal = grains.meanOrientation * sS;

% Normalize stress tensor
S=S./norm(S,'fro');
sigma=stressTensor(S);

SF_global=SchmidFactor(sSLocal,sigma);

%%%%%%%%%%%%%%%%% Analysis in crystal coordinate%%%%%%%%%%%%%%%%%%%%%%%
% Rotate stress tensor to local/crystal coordinates
sigmaLocal = inv(grains.meanOrientation)*sigma

sigmaLocal.M=sigmaLocal.M./norm(sigmaLocal.M,'fro');  % normalize 2nd time

SF_crystal = SchmidFactor(sS,sigmaLocal);

SF_crystal
SF_global

 

`
[SF.zip](https://github.com/mtex-toolbox/mtex/files/11316227/SF.zip)

**Comment by ralfHielscher** [comment, marked as answer]
Some comments:

1. Which version of MTEX are you using? The current version should perform the normalization already automatically.
2. The normalization you are using is not correct. You need to divide by the difference between the largest and smallest eigenvalue
3. Your zip file is corrupted.

Ralf.

**Reply by Ronit-Roy** [reply]
Hi Ralf,

Thanks a lot for your prompt response. It worked! The normalization I used earlier wasn't correct.
- I was using 5.1.1 version. 
- I have reuploaded the zip with the modified script for any future reference. 
Thank you once again for your kind help.

Best regards,
Ronit

[SF.zip](https://github.com/mtex-toolbox/mtex/files/11318990/SF.zip)
